autorest/client.go

```diff
@@ -262,3 +311,80 @@ func (c Client) ByInspecting() RespondDecorator {
 	}
 	return c.ResponseInspector
 }
+
+func withLogging() PrepareDecorator {
+	return func(p Preparer) Preparer {
+		return PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			// execute any remaining preparers before logging
+			r, err := p.Prepare(r)
+			if logLevel > logLevelNone {
+				b := &bytes.Buffer{}
+				fmt.Fprintf(b, "(%s) REQUEST: %s %s\n", time.Now().Format(logTimestamp), r.Method, r.URL.String())
+				// dump headers
+				for k, v := range r.Header {
+					if strings.ToLower(k) == "authorization" {
+						v = []string{"**REDACTED**"}
+					}
+					fmt.Fprintf(b, "%s: %s\n", k, strings.Join(v, ","))
+				}
+				ct := r.Header.Get(headerContentType)
+				if logLevel == logLevelFull && r.Body != nil && strings.Index(ct, mimeTypeOctetStream) == -1 {
+					// dump body
+					var body []byte
+					body, err = ioutil.ReadAll(r.Body)
+					if err == nil {
+						fmt.Fprintln(b, string(body))
+						if nc, ok := r.Body.(NopClosingReadSeeker); ok {
+							// rewind to the beginning
+							nc.Seek(0, io.SeekStart)
+						} else {
+							// recreate the body
+							r.Body = NopClosingReadSeeker{bytes.NewReader(body)}
+						}
+					} else {
+						fmt.Fprintf(b, "failed to read body: %v\n", err)
+					}
+				}
+				logFile.WriteString(b.String())
+				logFile.Sync()
```

Comment:
Other than this line, I'd recommend that you use the `log` package in the standard library.

Reply:
I had originally used the log package but I didn't see any way to flush the underlying stream.  Can it be done?

Reply:
I don't think so :/ I agree with your decision here. I was just trying to leave a comment here in case anybody else wondered why you didn't use it.

Reply:
Good point, I'll add a comment explaining why I used *os.File instead of a logger.